src/qn_model.jl

```diff
@@ -2,28 +2,30 @@ export QuasiNewtonModel, LBFGSModel, LSR1Model
 
 abstract type QuasiNewtonModel <: AbstractNLPModel end
 
-mutable struct LBFGSModel <: QuasiNewtonModel
-  meta :: NLPModelMeta
-  model :: AbstractNLPModel
+mutable struct LBFGSModel{M <: AbstractNLPModelMeta, T <: AbstractNLPModel} <: QuasiNewtonModel
+  meta :: M
+  model :: T
   op :: LBFGSOperator
 end
 
-mutable struct LSR1Model <: QuasiNewtonModel
-  meta :: NLPModelMeta
-  model :: AbstractNLPModel
+mutable struct LSR1Model{M <: AbstractNLPModelMeta, T <: AbstractNLPModel} <: QuasiNewtonModel
+  meta :: M
+  model :: T
   op :: LSR1Operator
 end
 
 "Construct a `LBFGSModel` from another type of model."
-function LBFGSModel(nlp :: AbstractNLPModel; kwargs...)
+function LBFGSModel(nlp :: T; kwargs...) where T <: AbstractNLPModel
```

Comment:
Sorry I think this PR got forgotten. I'm happy with all the changes but does this one really make a difference?

Reply:
I think I ran into this because I implemented the [multiple objective model meta](https://github.com/CiDAMO/MultObjNLPModels.jl/blob/main/src/meta.jl) by creating a new meta `<: AbstractNLPModelMeta`, and I couldn't call `unconstrained` on it.
I have since simply copied these functions over. I think this can be ignored for now. It will be relevant if we decide to change/extend meta in the future. Closing.